boa_interner/src/lib.rs

```diff
@@ -67,7 +67,9 @@
     clippy::missing_errors_doc,
```

Comment:
I'm reevaluating whether this design is worth, because maintaining alignment consumes more memory at the long run, whereas having two independent interners for each encoding consumes the same amount of memory for small and large amounts of strings.
After benchmarks I'll experiment with another design to see if the performance improves.

Reply:
Performance of this new implementation is hitting us significantly, everything that can be done in the future to improve this will be great :)